Create toolbar

hlyang397 opened the pull request:
### Updates
- list
- out
- updates
- here (and don't forget to link the issues)

### Demo screenshot or recording

### Review checklist (for reviewers only)
- [ ] Demos all features
- [ ] Documented/annotated
- [ ] Matches UI/UX specs
- [ ] Meets the [code style guide](https://github.com/carbon-design-system/carbon-charts/wiki/Code-style-guide)
- [ ] Accessible
- [ ] Mobile first (responsive)
- [ ] RTL support (bidirectional text)
- [ ] Performant (limited bloat)

hlyang397 committed:
feat: create toolbar components

- toolbar configuration
- zoom in/out icon
- overflow menu
- enable toolbar in zoom bar storybook

hlyang397 committed:
refactor: use constants in configuration

theiliad commented:
Is this PR in the wrong place? Looks like you made a PR to your own fork rather than the Charts repo @hlyang397

hlyang397 committed:
feat: create toolbar storybook demo group

- remove toolbar options in zoom bar demo group

hlyang397 committed:
refactor: toolbar button

- simplify css
- refactor toolbar button generation code

hlyang397 committed:
feat: disable toolbar button if it won't work

- allow button disabled state
- check min/max zoom domain for zoom in/out button

hlyang397 commented:
> Is this PR in the wrong place? Looks like you made a PR to your own fork rather than the Charts repo @hlyang397

@theiliad This is PR is for @JennChao to review internally in our team before creating a PR in Charts repo.
Sorry I didn't realize that this PR will request your review automatically.

Do you know how to disable this automatic review request in private repo ?

hlyang397 committed:
fix: title should use parent node to get max title width

hlyang397 committed:
feat: provide minZoomRatio option

Commit:
Merge branch 'master' into create-toolbar

theiliad commented:
> Do you know how to disable this automatic review request in private repo ?

no idea, it's probably the `CODEOWNERS` file that lists my github username

hlyang397 committed:
Merge branch 'master' into create-toolbar

hlyang397 committed:
feat  allow overflow menu item to be disabled

hlyang397 committed:
Merge branch 'master' into create-toolbar

# Conflicts:
#	packages/core/src/configuration.ts
#	packages/core/src/interfaces/components.ts
#	packages/core/src/services/zoom.ts

hlyang397 committed:
feat: combine toolbar icon and overflow menu item to controls

- create ToolbarControlTypes enum
- change configuration and interfaces
- decide how many controls are in button and in overflow menu
- change svg icon from 16x16 to 32x32 viewBox

hlyang397 committed:
Merge branch 'master' into create-toolbar

hlyang397 committed:
Merge branch 'master' into create-toolbar

hlyang397 committed:
fix: not import src/ in demo folder to avoid build failure

hlyang397 committed:
feat: support keyboard event for toolbar button

- make control focusable
- specify roles in control

hlyang397 committed:
refactor: fix scss style

hlyang397 committed:
refactor: change configuration name

- maxIcons => numberOfIcons
- controlsInOrder => controls

hlyang397 committed:
refactor: remove default ToolbarControls in configuration

hlyang397 committed:
Merge branch 'master' of github.com:carbon-design-system/carbon-charts into create-toolbar

hlyang397 committed:
feat: allow to use keyboard to control overflow menu

hlyang397 committed:
Merge branch 'master' into create-toolbar

hlyang397 committed:
fix: align toolbar to title component

hlyang397 committed:
Merge branch 'master' into create-toolbar

# Conflicts:
#	packages/core/src/configuration.ts
#	packages/core/src/services/zoom.ts

hlyang397 committed:
refactor: refactor statements and fix typos

hlyang397 committed:
refactor: change ToolbarControl.text field to optional

hlyang397 committed:
refactor: use getControlConfigByType() to reduce methods count

hlyang397 committed:
Merge branch 'master' into create-toolbar

hlyang397 committed:
refactor: minor changes per comments

- titlePostfix -> titleSuffix
- showOverflowMenu -> updateOverflowMenu
- VERTICAL_ALIGNMENT_Y -> Y_OFFSET

hlyang397 committed:
refactor: remove toolbar control text in demo

hlyang397 committed:
refactor: use d3.select to replace document.getElementById
